addons/sale_timesheet/models/project.py

```diff
@@ -474,6 +474,8 @@ def merge_profitability_data(a, b):
         for cost in profitability_items['costs']['data']:
             cost_id = cost['id']
             aal_cost = costs_dict.pop(cost_id, {})
+            if not aal_cost:
+                continue
```

Comment:
Indeed, it's better to avoid this change here, but it seems necessary because in the case where we only have one Mo in draft, therefore no analytic account, 'billed' and 'to_bill' will not be added in the cost:

https://github.com/odoo/odoo/pull/124714/files#diff-6068aa89213e5a8fdf9b6c258be8a98f0eea3b25386b8609fcf541e675fc673aR78-R83

So, when we want to modify their value here "cost['to_bill'] +=", a traceback will be raised:
https://github.com/odoo/odoo/pull/124714/files#diff-26cc107833e7deb17fb48bb9fa59f0b4d170d7091d89c194042cc7ca7e1d4074L477

If we decide to add "to_bill" anyway with a value of 0 to avoid this change, the action will not be displayed:

https://github.com/odoo/odoo/commit/b8ef0cc4f22827f684cf3361ee9fd526d27cb72a#diff-301941e7e0d04954a0acfbd79426af4ad6aeb74c3e135a75e02919f2cb77b22cR51

Reply:
Indeed, it's better to avoid this change here, but it seems necessary because in the case where we only have one Mo in draft, therefore no analytic account, 'billed' and 'to_bill' will not be added in the cost:

https://github.com/odoo/odoo/pull/124714/files#diff-6068aa89213e5a8fdf9b6c258be8a98f0eea3b25386b8609fcf541e675fc673aR78-R83

So, when we want to modify their value here "cost['to_bill'] +=", a traceback will be raised:
https://github.com/odoo/odoo/pull/124714/files#diff-26cc107833e7deb17fb48bb9fa59f0b4d170d7091d89c194042cc7ca7e1d4074L477

If we decide to add "to_bill" anyway with a value of 0 to avoid this change, the action will not be displayed:

https://github.com/odoo/odoo/commit/b8ef0cc4f22827f684cf3361ee9fd526d27cb72a#diff-301941e7e0d04954a0acfbd79426af4ad6aeb74c3e135a75e02919f2cb77b22cR51

Reply:
It is not valid, we should include the MO in draft instead of adding the category just to have the button.